cub/test/catch2_test_device_transform.cu

```diff
@@ -684,3 +684,68 @@ C2H_TEST("DeviceTransform::Transform function/output_iter return type not conver
   c2h::device_vector<C> reference(num_items, C{-43});
   CHECK(output == reference);
 }
+
+#if TEST_LAUNCH == 0
+
+template <int BlockThreads, int ItemsPerPthread, typename T>
+__global__ void fill_pdl_kernel(T* data, size_t n, T value)
+{
+  // we trigger the next kernel's launch very soon and wait a bit for it to spin up before starting to write. this way
+  // we try to expose the next kernel reading uninitialized data, if it contains a bug.
+  _CCCL_PDL_GRID_DEPENDENCY_SYNC();
+  _CCCL_PDL_TRIGGER_NEXT_LAUNCH();
+  __nanosleep(100'000); // must be enough to cover the launch overhead of the next kernel
```

Comment:
@gevtushenko this can successfully expose a missing `_CCCL_PDL_GRID_DEPENDENCY_SYNC` in the follow-up kernel. The key is to wait long enough for the next kernel to start running *and* make sure the problem size allows for CTAs of this and the follow-up kernel to start running together.